spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/startup/StartupTimeMetrics.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2012-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.metrics.startup;
+
+import java.lang.management.RuntimeMXBean;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.TimeGauge;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.boot.context.event.ApplicationStartedEvent;
+import org.springframework.context.ApplicationEvent;
+import org.springframework.context.event.SmartApplicationListener;
+
+/**
+ * Binds application startup metrics in response to {@link ApplicationStartedEvent} and
+ * {@link ApplicationReadyEvent}.
+ *
+ * @author Chris Bono
+ * @since 2.6.0
+ */
+public class StartupTimeMetrics implements SmartApplicationListener {
+
+	private final MeterRegistry meterRegistry;
+
+	private final RuntimeMXBean runtimeMXBean;
+
+	private final String applicationStartedTimeMetricName;
+
+	private final String applicationReadyTimeMetricName;
+
+	private final String applicationReadyJvmTimeMetricName;
+
+	private final Iterable<Tag> tags;
+
+	public StartupTimeMetrics(MeterRegistry meterRegistry, RuntimeMXBean runtimeMXBean,
+			String applicationStartedTimeMetricName, String applicationReadyTimeMetricName,
```

Comment:
I thought of introducing a builder, however all of these args are required (except tags) and the defaults are managed in the config properties so it would not buy us much.